nucypher-core-python/src/lib.rs

```diff
@@ -773,11 +785,11 @@ impl NodeMetadataPayload {
             .transpose()?;
         Ok(Self {
             backend: nucypher_core::NodeMetadataPayload {
-                staking_provider_address: nucypher_core::Address::new(&staking_provider_address),
+                staking_provider_address: staking_provider_address.backend,
                 domain: domain.to_string(),
                 timestamp_epoch,
-                verifying_key: verifying_key.backend,
-                encrypting_key: encrypting_key.backend,
+                verifying_key: *verifying_key.as_ref(),
```

Comment:
`.as_ref()` gives a reference to the inner object of the newtype (`umbral_pre::PublicKey`, as opposed to the outer `umbral_pre::bindings_python::PublicKey`), then we need to make a copy of it to pass to the method. You could call `.clone()`, and that would compile, but `clippy` complains about it because `umbral_pre::PublicKey` implements `Copy`, and it recommends you to just use that. The dereferencing turns a reference into an actual object, effectively copying it to the stack.